### DISCUSSION TITLE
Add ability to snap windows to screen corners

### DISCUSSION THREAD

**Discussion by markquiriconi** [f1multiviewer/issue-tracker#3] [Feature requests] [answered]
Hello,

It would be great to have the ability to snap windows to the corners of your screen to avoid pixel gaps and also make arranging feeds way faster. The Race Control app had this where you could click an arrow for Top Left, Top Right, Bottom Left and Bottom Right and you could fill a screen (as it would size them to fill the screen) fully with four feeds very quickly rather than having to drag and try to get the alignment the same on four video feeds

Below is the way it was laid out on race control, but I suppose just having it as some sort of menu option when you right click is fine as well.
![race control](https://user-images.githubusercontent.com/49886918/173076567-02818250-1c99-4f30-955d-89cbabea3f50.png)
.

**Comment by aeleos** [comment]
I would also like this. I think it might also might be nice to snap to some sort of other grid (or even other f1multiviewer windows). Maybe you could even setup a couple input parameters to a CSS grid to allow more control.

**Reply by Jurtaani** [reply]
While I am not against this idea, Windows 10 already has corner snapping features built-in.
If you drag the window to the middle on each sides, it will fill exactly half of the screen but if you instead do that to the either top or bottom corner of the screen, then it will resize the window with 1/4 sized player window.

Yes having race control styled arrows or something like that would be nice, but for the time being the windows default is ok enough. 
If you are using windows that is.

**Comment by Petarkco** [comment]
To add on to this I noticed that when reloading a saved layout where the windows were snapped to the edges that F1MV will open them with gaps around each window 
![image](https://user-images.githubusercontent.com/25685344/173171099-9d6d5cf1-3f99-4224-a61f-4ddb7b797433.png)

This is on Windows 11

**Reply by f1multiviewer** [reply]
To avoid this, you may have to turn off "Maintain aspect ratio"

**Comment by f1multiviewer** [comment, marked as answer]
This is enabled now!
